framework/db/BaseActiveRecord.php

```diff
@@ -412,12 +406,24 @@ public function hasOne($class, $link)
      */
     public function hasMany($class, $link)
     {
+        return $this->newRelationQuery($class, $link, true);
+    }
+
+    /**
+     * Creates a query for `has-one` and `has-many` relation.
+     * @param string $class the class name of the related record
+     * @param array $link the primary-foreign key constraint
+     * @param boolean whether this query represents a relation to more than one record.
+     * @return ActiveQueryInterface the relational query object.
+     */
+    protected function newRelationQuery($class, $link, $multiple)
```

Comment:
this should be called `createRelationQuery` and if we want to be consistent even `createLinkQuery`